### DISCUSSION TITLE
Improve Display Of Step-Linked Ingredients

### DISCUSSION THREAD

**Discussion by royaltongue** [mealie-recipes/mealie#4474] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Currently, there doesn't seem to be a way to see which steps have which ingredients linked to them without going into "Cook Mode". Further, even when in Cook Mode, the linked ingredients are displayed in small text at the bottom of the step, which I think can be very easy to not see/miss for less-knowledgable users.

### Please provide a concise description of the feature that would resolve your issue.

My suggestions:
1) Show linked ingredients on the default recipe page, without going into Cook Mode
2) Make linked ingredients more obvious and prominent
3) Potentially add a checklist of linked ingredients per step? Tandoor does something similar, which you can see here: 
![image](https://github.com/user-attachments/assets/b5e2eeb1-6270-4901-8131-d07e75c994bf)

### Please consider and list out some caveats or tradeoffs made in your design decision

N/A?

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Kuchenpirat** [comment]
I feel like this would be an improvement to our current system 👍

Also this is related to #4464 which is a PR that implements a similar layout for a cookmode when ingredients are not linked

**Comment by codetakki** [comment]
i have update #4464 to include display ingredients in sections based on its step, with a "other" for none linked.
